internal/controllers/phase_ensure_operator_group.go

```diff
@@ -37,6 +37,8 @@ func (r *AddonReconciler) ensureOperatorGroup(
 		desiredOperatorGroup.Spec.TargetNamespaces = []string{targetNamespace}
 	}
 
+	// TODO - if the OG already exists, and is already owned by this controller, will this error?
+	// quickly digged in SetControllerReference and did not see any check for this
```

Comment:
`SetControllerReference` errors if the object already has an owner, but we call it on the desired object that we are building in line No.29, so this will always succeed.
We don't currently "adopt" objects that are happen to be already present, maybe we have to do this for the transition from the current flow :thinking: 

Created a ticket to check about that:
https://issues.redhat.com/browse/MTSRE-116

Reply:
I feel it should be fine to take ownership of existing namespaces.

And the idea I had was:
1. if the namespace does not exist, simply create
2. if it exists, then try and call `SetControllerReference` on the existing (not desired) object, and go from there...

Reply:
you are right, not sure why I put that comment there , removed